logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ClearConversationContent.kt

```diff
@@ -59,29 +60,31 @@ interface ClearConversationContentUseCase {
 internal class ClearConversationContentUseCaseImpl(
     private val clearConversationContent: ClearConversationContent,
     private val clientRepository: ClientRepository,
-    private val conversationRepository: ConversationRepository,
     private val messageSender: MessageSender,
     private val selfUserId: UserId,
+    private val selfConversationIdProvider: SelfConversationIdProvider
 ) : ClearConversationContentUseCase {
 
     override suspend fun invoke(conversationId: ConversationId): ClearConversationContentUseCase.Result =
         clearConversationContent(conversationId).flatMap {
             clientRepository.currentClientId().flatMap { currentClientId ->
-                val regularMessage = Message.Regular(
-                    id = uuid4().toString(),
-                    content = MessageContent.Cleared(
-                        unqualifiedConversationId = conversationId.value,
-                        conversationId = conversationId,
-                        time = Clock.System.now()
-                    ),
-                    conversationId = conversationRepository.getSelfConversationId(),
-                    date = Clock.System.now().toString(),
-                    senderUserId = selfUserId,
-                    senderClientId = currentClientId,
-                    status = Message.Status.PENDING,
-                    editStatus = Message.EditStatus.NotEdited,
-                )
-                messageSender.sendMessage(regularMessage)
+                selfConversationIdProvider().flatMap { selfConversationId ->
+                    val regularMessage = Message.Regular(
+                        id = uuid4().toString(),
+                        content = MessageContent.Cleared(
+                            unqualifiedConversationId = conversationId.value,
+                            conversationId = conversationId,
+                            time = Clock.System.now()
+                        ),
+                        conversationId = selfConversationId,
```

Comment:
Why do we need to set the `conversationId` with the `selfConversationId` for clearing the content of another convo? is it because we use it as a "bridge" between all our clients?

Reply:
selfConversation is used to sync between user clients, so the event to clear a conversation content is sent to self conversation, I will add some comments to make it clear